File: src/main/java/com/quid/spring3/restClient/RestClientConfig.java
package com.quid.spring3.restClient;

import lombok.extern.slf4j.Slf4j;
import org.springframework.web.client.RestClient;

@Slf4j
public class RestClientConfig {

    public RestClient restClient() {
        return RestClient.builder()
            .baseUrl("https://www.naver.com")
            .requestInterceptor((request, body, execution) -> {
                log.info("Request: {}", request);
                return execution.execute(request, body);
            })
            .build();
    }



}

File: src/main/java/com/quid/resilience4j/RestClientConfig.kt
package com.quid.resilience4j

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.client.RestClient

@Configuration
class RestClientConfig {

    @Bean
    fun restclient(): RestClient {
        return RestClient.builder()
            .baseUrl("http://localhost:8080/test/random")
            .build()
    }
}
